lib/child_process.js

```diff
@@ -425,13 +425,15 @@ function normalizeSpawnArguments(file, args, options) {
   }
 
   // Validate the uid, if present.
-  if (options.uid != null && !Number.isInteger(options.uid)) {
-    throw new ERR_INVALID_ARG_TYPE('options.uid', 'integer', options.uid);
+  if (options.uid != null &&
+      (!Number.isInteger(options.uid) || !isUint32(options.uid))) {
```

Comment:
Is the `Number.isInteger()` check still needed? Also, don't we want int32 checks to match the C++ checks?

Reply:
It will look kind of strange (especially if it's an object {} >>> 1 and {} | 0) but I guess it is redundant.

I actually thought maybe it's worth changing them too, though I don't have enough knowledge of the topic. Linux usage and googling tell me that uid/gid must be non-negative, though maybe some of the systems we support actually support negative uid/gid. Therefore I'd also like some input on this.
In the end, I can always change those to isint32 and a topic of changing them to uint checks may be resolved in a separate PR.

Reply:
After some more googling OpenBSD seem to use -2 as internal "nobody" still.
For now, I'll change it to Int32 and we can discuss this later.